src/components/Room/Room.test.tsx

```diff
@@ -48,63 +48,111 @@ describe('the Room component', () => {
     );
-    expect(mockSetIsGridModeActive).not.toBeCalled();
+    expect(mockSetIsGridViewActive).not.toBeCalled();
   });
 
-  it('should deactivate grid mode when a remote participant shares their screen', () => {
+  it('should deactivate grid view when a remote participant shares their screen', () => {
     const { rerender } = renderHook(
       ({ screenShareParticipant }) =>
-        useSetCollaborationViewOnScreenShare(
+        useSetPresentationViewOnScreenShare(
           screenShareParticipant,
           { localParticipant: {} } as any,
-          mockSetIsGridModeActive
+          mockSetIsGridViewActive,
+          true
         ),
       { initialProps: { screenShareParticipant: undefined } }
     );
-    expect(mockSetIsGridModeActive).not.toBeCalled();
+    expect(mockSetIsGridViewActive).not.toBeCalled();
     rerender({ screenShareParticipant: {} } as any);
-    expect(mockSetIsGridModeActive).toBeCalledWith(false);
+    expect(mockSetIsGridViewActive).toBeCalledWith(false);
   });
 
-  it('should not deactivate grid mode when the local participant shares their screen', () => {
+  it('should reactivate grid view when screenshare ends if grid view was active before another participant started screensharing', () => {
+    jest.spyOn(React, 'useRef').mockReturnValue({
+      current: true,
+    });
+
+    const { rerender } = renderHook(
+      ({ screenShareParticipant }) =>
+        useSetPresentationViewOnScreenShare(
+          screenShareParticipant,
+          { localParticipant: {} } as any,
+          mockSetIsGridViewActive,
+          true
+        ),
+      { initialProps: { screenShareParticipant: undefined } }
+    );
+    expect(mockSetIsGridViewActive).not.toBeCalled();
+    // screenshare starts
+    rerender({ screenShareParticipant: {} } as any);
+    expect(mockSetIsGridViewActive).toBeCalledWith(false);
+    // screenshare ends
+    rerender({ screenShareParticipant: {} } as any);
+    expect(mockSetIsGridViewActive).toBeCalledWith(true);
+  });
+
+  it('should not reactivate grid view when screenshare ends if grid view was active before another participant started screensharing', () => {
+    jest.spyOn(React, 'useRef').mockReturnValue({
+      current: false,
+    });
+
+    const { rerender } = renderHook(
+      ({ screenShareParticipant }) =>
+        useSetPresentationViewOnScreenShare(
+          screenShareParticipant,
+          { localParticipant: {} } as any,
+          mockSetIsGridViewActive,
+          true
+        ),
+      { initialProps: { screenShareParticipant: undefined } }
+    );
+    expect(mockSetIsGridViewActive).not.toBeCalled();
+    rerender({ screenShareParticipant: {} } as any);
+    expect(mockSetIsGridViewActive).toBeCalledWith(false);
+    // mockSetIsGridViewActive should only be called once with "false" since we're not reactivating grid mode
+    expect(mockSetIsGridViewActive).toBeCalledTimes(1);
+  });
+
```

Comment:
Once more test could be nice to test that presentation mode is not activated if the user switches to grid mode during a screenshare:

```
  it('should not activate presentation view when screenshare ends if presentation view was active before screensharing, but the user switched to grid view during the screenshare', () => {
    const screenShareParticipant = {};
    const room = { localParticipant: {} } as any;

    const { rerender } = renderHook(
      ({ screenShareParticipant, isGridViewActive }) =>
        useSetCollaborationViewOnScreenShare(screenShareParticipant, room, mockSetIsGridViewActive, isGridViewActive),
      { initialProps: { screenShareParticipant: undefined, isGridViewActive: false } }
    );

    expect(mockSetIsGridViewActive).not.toBeCalled();

    // start screenshare
    rerender({ screenShareParticipant, isGridViewActive: false } as any);
    expect(mockSetIsGridViewActive).toBeCalledWith(false);

    // enable grid mode
    rerender({ screenShareParticipant, isGridViewActive: true } as any);

    // stop screenshare
    rerender({ screenShareParticipant: undefined, isGridViewActive: true } as any);

    // mockSetIsGridViewActive should only be called once with "false" since we're not reactivating grid mode
    expect(mockSetIsGridViewActive).toBeCalledTimes(1);
  });
```

I wouldn't mind a more concise description if you can think of one 😆.

Also, note that I saved the room object and screenshare object to their own variables. This is so hook rerenders happen with the same object references as the previous render. This prevents the useEffects from running unnecessarily. This wasn't needed in the previous tests, but it's needed in this one.

Reply:
Removed a couple of words but couldn't really come up with anything much shorter 😢